flycheck.el

```diff
@@ -6814,7 +6814,13 @@ See URL `https://github.com/commercialhaskell/stack'."
   (lambda (errors)
     (flycheck-sanitize-errors (flycheck-dedent-error-messages errors)))
   :modes (haskell-mode literate-haskell-mode)
-  :next-checkers ((warning . haskell-hlint)))
+  :next-checkers ((warning . haskell-hlint))
+  :predicate (lambda ()
+               (let ((fs (mapcar (lambda (f)
+					(and f (locate-dominating-file (buffer-file-name) f)))
```

Comment:
Please also be aware that `locate-dominating-file` can also take a function as second argument.  Probably you could use this to check for all files at once, thus saving a couple of directory traversals.